drakrun/drakrun/analyzer.py

```diff
@@ -0,0 +1,554 @@
+from stat import S_ISREG, ST_CTIME, ST_MODE, ST_SIZE
+from typing import Any, Dict, List, Optional, Tuple
+
+import magic
+
+from .lib.config import load_config
+from .lib.drakpdb import dll_file_list
+from .lib.injector import Injector
+from .lib.install_info import InstallInfo
+from .lib.networking import (
+    delete_vm_network,
+    setup_vm_network,
+    start_dnsmasq,
+    start_tcpdump_collector,
+)
+from .lib.paths import ETC_DIR, PROFILE_DIR, RUNTIME_FILE
+from .lib.sample_startup import get_sample_entrypoints, get_sample_startup_command
+from .lib.storage import get_storage_backend
+from .lib.util import RuntimeInfo, graceful_exit
+from .lib.vm import VirtualMachine
+
+log = logging.getLogger(__name__)
+
+
+class UnretryableAnalysisError(Exception):
+    """
+    Analysis error that is not intermittent and should not lead to the retry of the analysis
+    """
+
+
+class DrakvufVM:
+    def __init__(self, vm_id: int):
+        self.vm_id = vm_id
+        self.install_info = InstallInfo.load()
+        self.storage_backend = get_storage_backend(self.install_info)
+        self.runtime_info = RuntimeInfo.load(RUNTIME_FILE)
+        self.kernel_profile_path = os.path.join(PROFILE_DIR, "kernel.json")
+        self.vm = VirtualMachine(self.storage_backend, vm_id)
+        self.injector = Injector(
+            self.vm.vm_name, self.runtime_info, self.kernel_profile_path
+        )
+
+
+def prepare_output_dir(output_dir: pathlib.Path):
+    output_dir.mkdir(parents=True, exist_ok=True)
+    if any(output_dir.iterdir()):
+        raise RuntimeError("Output directory is not empty")
+    (output_dir / "dumps").mkdir()
+    (output_dir / "ipt").mkdir()
+
+
+@dataclasses.dataclass
+class AnalysisOptions:
+    sample_path: pathlib.Path
+    vm_id: int
+    output_dir: pathlib.Path
+    plugins: List[str]
+    timeout: int = 600
+    hooks_path: pathlib.Path = pathlib.Path(ETC_DIR) / "hooks.txt"
+    start_command: Optional[str] = None
+    extension: Optional[str] = None
+    sample_filename: Optional[str] = None
+    dns_server: Optional[str] = None
+    out_interface: Optional[str] = None
+    net_enable: bool = True
+    anti_hammering_threshold: int = 0
+    syscall_filter: Optional[str] = None
+    raw_memory_dump: bool = False
+
+
+def filename_for_task(options: AnalysisOptions) -> Tuple[str, str]:
+    """
+    Return a tuple of (filename, extension) for a given task.
+    This depends on the content magic, "extension" and "file_name" options.
+    """
+    extension = options.extension
+    if not extension:
+        magic_output = magic.from_file(options.sample_path)
+        if "(DLL)" in magic_output:
+            extension = "dll"
+        else:
+            extension = "exe"
+    # Make sure the extension is lowercase
+    extension = extension.lower()
+    # TODO: sample_filename should already contain extension or be fixed
+    #       to the proper extension
+    file_name = (options.sample_filename or "malwar") + f".{extension}"
```

Comment:
maybe we should call the files differently? For example randomize them if we don't have a better name. :thinking: I know this is here since the beginning, but this is pretty trivial to bypass.